src/Components/test/testassets/Components.TestServer/RazorComponents/Pages/Forms/FormWithEnctypeAndSubmitButtonWithFormenctype.razor

```diff
@@ -0,0 +1,39 @@
+﻿@page "/forms/form-with-enctype-and-submit-button-with-formenctype/{enctype?}/{formenctype?}"
+@using Microsoft.AspNetCore.Components.Forms
+@inject IHttpContextAccessor HttpContextAccessor
+
+<h1>Form with enctype and Submit Button with formenctype</h1>
+
+<form id="form" data-enhance method="post" enctype="@Enctype" @onsubmit="Submitted" @formname="someform">
+    <AntiforgeryToken />
+    <input id="submit-button" type="submit" formenctype="@Formenctype" />
+</form>
+
+@if (_submitted)
+{
+    <p>Http request ContentType: <span id="content-type">@_contentType</span></p>
+}
+
+@code {
+    bool _submitted = false;
+    private string? _contentType;
+
+    [Parameter] public string? Enctype { get; set; }
+    [Parameter] public string? Formenctype { get; set; }
+
+    public void Submitted()
+    {
+        _submitted = true;
+
+        var httpContext = HttpContextAccessor.HttpContext;
+        _contentType = httpContext?.Request.ContentType;
+    }
+
+    protected override void OnParametersSet()
+    {
+        Enctype = Enctype?.Replace('%', '/');
+        Formenctype = Formenctype?.Replace('%', '/');
+
+        base.OnParametersSet();
+    }
```

Comment:
As per comment above, hopefully this `OnParametersSet` override can be removed entirely. Also FYI it's not necessary to call `base` methods from Blazor lifecycle methods.